File: 3264_final_array_state_after_k_multiplication_operations_I/getFinalState.ts
function getFinalState(
  nums: number[],
  k: number,
  multiplier: number
): number[] {
  for (let i = 0; i < k; i++) {
    let idx = getMinIdx(nums);
    nums[idx] *= multiplier;
  }

  return nums;
}

function getMinIdx(nums: number[]): number {
  let min = nums[0];
  let idx = 0;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] < min) {
      min = nums[i];
      idx = i;
    }
  }

  return idx;
}

File: 3264_final_array_state_after_k_multiplication_operations_I/getFinalState.go
package finalarraystateafterkmultiplicationoperationsi

func getFinalState(nums []int, k int, multiplier int) []int {
	for range k {
		idx := getMinIdx(nums)
		nums[idx] *= multiplier
	}

	return nums
}

func getMinIdx(nums []int) int {
	idx := 0
	min := nums[idx]

	for i, num := range nums {
		if num < min {
			min = num
			idx = i
		}
	}

	return idx
}
